src/strategies/sound-protocol-call-tokenuri/transformer.mjs

```diff
@@ -0,0 +1,92 @@
+// @format
+import { decodeParameters } from "eth-fun";
+import logger from "../../logger.mjs";
+
+export const name = "sound-protocol-call-tokenuri";
+export const version = "0.1.0";
+
+const resultKey = "tokenURI";
+const log = logger(name);
+
+const extractTokenId = (tokenURI) => parseInt(tokenURI.split("/")[3]);
+
+export function onClose() {
+  return;
+}
+
+export function onError(error) {
+  log(error.toString());
+  throw error;
+}
+
+export function onLine(line) {
+  let obj;
+
+  try {
+    obj = JSON.parse(line);
+  } catch (err) {
+    log(err.toString());
+    return;
+  }
+
+  let { results: tokenURIs } = obj;
+  let result = [];
+  try {
+    tokenURIs = tokenURIs.filter(
+      (tokenURI) => tokenURI.split("/").length === 4
+    );
+  } catch (err) {
+    log(err.toString());
+    return;
+  }
+
+  try {
+    tokenURIs.sort((a, b) => extractTokenId(a) - extractTokenId(b));
```

Comment:
Is this for catching errors in extractTokenId? I'm not sure if it is a good idea to catch errors here. 

A general rule for catching errors is

- if we find a condition that the program cannot be recovered from, then we want to throw an error that crashes the entire program
- if we find a condition that the strategy cannot recover from the we should stop executing the strategy
- if we find a condition that the strategy cannot recover from but that we may just e.g. lose one tokenURI in the process, then we may just silently error and move on to the next tokenURI
- if we find a condition that the strategy can recover from, then we should recover from it.

Does that make sense?

Reply:
Okay, it makes sense